web_widget_text_markdown/static/src/js/web_widget_text_markdown.js

```diff
@@ -0,0 +1,101 @@
+/* global marked */
+/* Copyright 2014 Sudokeys <http://www.sudokeys.com>
+ * Copyright 2017 Komit - <http:///komit-consulting.com>
+ * Copyright 2019 Alexandre Díaz - <dev@redneboa.es>
+ * License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). */
+odoo.define("web_widget_text_markdown.FieldTextMarkDown", function (require) {
+    'use strict';
+
+    var basic_fields = require('web.basic_fields');
+    var field_registry = require('web.field_registry');
+    var core = require('web.core');
+
+    var _t = core._t;
+    var LIBS_PATH = '/web_widget_text_markdown/static/src/lib';
+
+
+    var FieldTextMarkDown = basic_fields.FieldText.extend({
+        jsLibs: [
+            LIBS_PATH + '/marked.js',
+            LIBS_PATH + '/dropzone.js',
+            LIBS_PATH + '/bootstrap-markdown.js',
+        ],
+        cssLibs: [
+            LIBS_PATH + '/bootstrap-markdown.min.css',
+        ],
+
+        _getValue: function () {
+            return this.$markdown.getContent();
+        },
+
+        _prepareInput: function () {
+            var $result = this._super.apply(this, arguments);
+            _.defer(function ($elm) {
+                var markdownOpts = {
+                    autofocus: false,
+                    savable: false,
+                    dropZoneOptions: {
+                        paramName: 'ufile',
+                        url: '/web_widget_text_markdown/upload_attachment',
+                        acceptedFiles: 'image/*',
+                        params: {
+                            csrf_token: core.csrf_token,
+                            session_id: this.getSession().override_session,
+                            callback: '',
+                            model: this.model,
+                            id: this.res_id,
```

Comment:
watch out, this is not a real ID on record create. Either we handle it on final create/write or we state clearly in the readme that we cannot use image upload on create.
Am I wrong?